### PR TITLE
Permission issue | Nullable handler | UiThread issue

### PR DESCRIPTION
**Pull request by alisherbu**
I have done everything myself to make my app work properly. I didn't try to understand all and optimize. For now everything works fine. UiThread issue fixed as you said @rohitsangwan01 (Don't pay attention branch name🙈)

### CODE REVIEW

File: android/src/main/kotlin/com/rohit/ble_peripheral/BlePeripheralPlugin.kt [hunk 1]
```diff
@@ -173,8 +173,6 @@ class BlePeripheralPlugin : FlutterPlugin, BlePeripheralChannel, ActivityAware,
     private fun validatePermission(): Boolean {
         val permissionsList = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
             arrayOf(
-                Manifest.permission.BLUETOOTH_ADMIN,
-                Manifest.permission.BLUETOOTH,
                 Manifest.permission.BLUETOOTH_SCAN,
```
Comment (rohitsangwan01) [inline android/src/main/kotlin/com/rohit/ble_peripheral/BlePeripheralPlugin.kt:176]: @alisherbu are you sure we don't need these permissions ?
also i haven't tested if we really need BluetoothScan permission
BluetoothPermissions Details are here: https://developer.android.com/guide/topics/connectivity/bluetooth/permissions

Comment (alisherbu) [reply android/src/main/kotlin/com/rohit/ble_peripheral/BlePeripheralPlugin.kt:176]: Yeap, these are legacy permissions for older devices. https://developer.android.com/guide/topics/connectivity/bluetooth/permissions

Comment (alisherbu) [reply android/src/main/kotlin/com/rohit/ble_peripheral/BlePeripheralPlugin.kt:176]: I think so. I will test BluetoothScan permission later

Comment (rohitsangwan01) [reply android/src/main/kotlin/com/rohit/ble_peripheral/BlePeripheralPlugin.kt:176]: @alisherbu i would suggest to revert the Permission changes, i was also planning to recheck the permissions later
And we can merge rest of the changes for now

Comment (alisherbu) [reply android/src/main/kotlin/com/rohit/ble_peripheral/BlePeripheralPlugin.kt:176]: When I validate Manifest.permission.BLUETOOTH, OS(Samsung Android 12) always returns false (not granted). So I had to remove these

Comment (rohitsangwan01) [reply android/src/main/kotlin/com/rohit/ble_peripheral/BlePeripheralPlugin.kt:176]: @alisherbu for now you can add all these permissions like in example app, and we can work with the permissions in new pr

Comment (Bennik2000) [reply android/src/main/kotlin/com/rohit/ble_peripheral/BlePeripheralPlugin.kt:176]: On my google pixel device this check also fails every time even if all permissions are granted. I tried another device (an older OnePlus) and there it works.

Comment (rohitsangwan01) [reply android/src/main/kotlin/com/rohit/ble_peripheral/BlePeripheralPlugin.kt:176]: @Bennik2000 have you tried example app on main branch

Comment (Bennik2000) [reply android/src/main/kotlin/com/rohit/ble_peripheral/BlePeripheralPlugin.kt:176]: The example indeed works and the device gets discovered. However I get the following log `Need BLUETOOTH PRIVILEGED permission: Neither user 10704 nor current process has android.permission.BLUETOOTH_PRIVILEGED` but I guess this is a different issue.

Comment (rohitsangwan01) [reply android/src/main/kotlin/com/rohit/ble_peripheral/BlePeripheralPlugin.kt:176]: @Bennik2000 Seems like you are also facing [this](https://github.com/rohitsangwan01/ble_peripheral/issues/1#issuecomment-1709635050) issue with the permissions

**Comment by mglsj** [comment]
@rohitsangwan01 why is this pull request not yet merged?

**Comment by rohitsangwan01** [comment]
> @rohitsangwan01 why is this pull request not yet merged?

i wanted to check the required permissions for android before merge, but now am thinking of removing the logic of permissions from sdk, and let the developers handle them
Or another option is to define the required permissions and all within plugin, and developers don't have to add anything from their end, but for this i need to test, which permissions are exactly required

**Comment by mglsj** [comment]
> > @rohitsangwan01 why is this pull request not yet merged?
> 
> i wanted to check the required permissions for android before merge, but now am thinking of removing the logic of permissions from sdk, and let the developers handle them Or another option is to define the required permissions and all within plugin, and developers don't have to add anything from their end, but for this i need to test, which permissions are exactly required

I think you should leave the permissions part to the developer.  Most devs already use "permission_handler" in their projects which makes its very easy to manage permissions.

**Comment by rohitsangwan01** [comment]
@alisherbu Thank you so much for the pr, plugin Api's are changed now, so this pr will be having merge conflicts, and i applied these fixes in new release https://github.com/rohitsangwan01/ble_peripheral/pull/7
if still there is something to fix, you can push a new pr

**Pull request closed without merging**